node_ops/task_manager_module.cc

```diff
@@ -31,4 +42,501 @@ std::string node_ops_task_impl::type() const {
     return fmt::format("{}", _reason);
 }
 
+join_token_ring_task_impl::join_token_ring_task_impl(tasks::task_manager::module_ptr module,
+        std::string entity,
+        service::storage_service& ss,
+        sharded<db::system_distributed_keyspace>& sys_dist_ks,
+        sharded<service::storage_proxy>& proxy,
+        std::unordered_set<gms::inet_address> initial_contact_nodes,
+        std::unordered_set<gms::inet_address> loaded_endpoints,
+        std::unordered_map<gms::inet_address, sstring> loaded_peer_features,
+        std::chrono::milliseconds delay) noexcept
+    : node_ops_task_impl(std::move(module), tasks::task_id::create_random_id(), ss.get_task_manager_module().new_sequence_number(),
+        "node", std::move(entity), tasks::task_id::create_null_id(), streaming::stream_reason::bootstrap, ss)
+    , _sys_dist_ks(sys_dist_ks)
+    , _proxy(proxy)
+    , _initial_contact_nodes(std::move(initial_contact_nodes))
+    , _loaded_endpoints(std::move(loaded_endpoints))
+    , _loaded_peer_features(std::move(loaded_peer_features))
+    , _delay(delay)
+{}
+
+gms::inet_address join_token_ring_task_impl::get_broadcast_address() const {
+    return _ss.get_broadcast_address();
+}
+
+bool join_token_ring_task_impl::should_bootstrap() {
+    return _ss.should_bootstrap();
+}
+
+bool join_token_ring_task_impl::is_replacing() {
+    return _ss.is_replacing();
+}
+
+locator::token_metadata_ptr join_token_ring_task_impl::get_token_metadata_ptr() const noexcept {
+    return _ss.get_token_metadata_ptr();
+}
+const locator::token_metadata& join_token_ring_task_impl::get_token_metadata() const noexcept {
+    return _ss.get_token_metadata();
+}
+
+future<> join_token_ring_task_impl::run() {
+    auto& _sys_ks = _ss._sys_ks;
```

Comment:
nit: since you move the code anyway, keeping the name aliases doesn't make any sense